src/model/non-linear-model/NonLinearRegression.ts

```diff
@@ -0,0 +1,111 @@
+import { Rank, Tensor, scalar, variable, Tensor1D, RecursiveArray, Variable, Scalar, sub, add, gather, stack, tidy, dispose } from '@tensorflow/tfjs-core';
```

Comment:
I think we need to unify our naming convention, see https://github.com/imgcook/datacook/blob/ce426da8c927d71fb5fb5425653603447c8a55be/src/model/linear-model/linear-regression.ts.

Reply:
The inconsistency is due to the method used to optimize non linear model is not stochastic gradient descent